### PR TITLE
Updated help for ASRRMcmdlets

### PR DESCRIPTION
**Pull request by CFreemanwa**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by markcowl** [comment]
@CFreemanwa Please follow the instructions here:  for regenerating the dll-Help.xml file associated with these changes.https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md

LMK if there are any issues with this.

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@avneeshrai Can you review?

**Comment by markcowl** [comment]
@CFreemanwa It looks liek the new help is missing for cmdlet Get-AzureRmSiteRecoveryNetwork.  Can you make sure this is added?

**Comment by CFreemanwa** [comment]
@markcowl I'm not sure why it wasn't included. It is in the list of updated files.

**Comment by markcowl** [comment]
@CFreemanwa It apprently didn't make it in to the dll-Help.xml update, can you take a look?

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ Accept wildcard characters: False
 ```
 
 ### -Group
-Specifies a Site Recovery plan group.
+An instance of the ASRRecoveryPlanGroup which the plan belongs to.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md]: Its actually the other way around. Recovery Plan Groups are groupings under the recovery plan. There could be multiple Recovery plan groups in a recovery plan. I'd prefer we call it "Recovery Plan" and "Recovery Plan Group" as those are the nouns for these constructs. The class definition has ASR prefixed to distinguish this as an Azure Site Recovery (ASR) construct.

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md [hunk 2]
```diff
@@ -143,7 +145,7 @@ Accept wildcard characters: False
 ```
 
 ### -RemoveProtectedEntities
-Specifies an array of protected entities.
+An instance of an ASRProtectionEntity array of protected entities to remove.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md]: An instance of an array seems confusing. This is an array of ASRProtectionEntity instances to remove

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Start-AzureRmSiteRecoveryTestFailoverJob.md [hunk 3]
```diff
@@ -191,6 +191,8 @@ Accept wildcard characters: False
 ```
 
 ### -ReplicationProtectedItem
+The item to be protected.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Start-AzureRmSiteRecoveryTestFailoverJob.md]: The ASR replication protected item to be test failed over

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Start-AzureRmSiteRecoveryUnplannedFailoverJob.md [hunk 4]
```diff
@@ -141,6 +141,8 @@ Accept wildcard characters: False
 ```
 
 ### -ReplicationProtectedItem
+The item to be protected.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Start-AzureRmSiteRecoveryUnplannedFailoverJob.md]: The ASR replication protected item to be failed over

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Update-AzureRmSiteRecoveryPolicy.md [hunk 5]
```diff
@@ -20,12 +22,16 @@ Update-AzureRmSiteRecoveryPolicy -Policy <ASRPolicy> [-ReplicationMethod <String
 ```
 
 ## DESCRIPTION
+The **Update-AzureRmSiteRecoveryPolicy** updates the Site Recovery policy specified by the Policy parameter. The policy parameters control the various configuration setting for Site Recovery.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Update-AzureRmSiteRecoveryPolicy.md]: settings

File: src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md [hunk 6]
```diff
@@ -128,7 +130,7 @@ Accept wildcard characters: False
 ```
 
 ### -AddProtectedEntities
-Specifies an array of protected entities to add.
+An instance of an ASRProtectionEntity array of protected entities to add.
```
Comment (bsiva) [inline src/ResourceManager/SiteRecovery/Commands.SiteRecovery/help/Edit-AzureRmSiteRecoveryRecoveryPlan.md]: An instance of an array seems confusing. This is an array of ASRProtectionEntity instances to add

**Comment by cormacpayne** [comment]
@CFreemanwa Hey Charles, quick update: I'm currently working on a pull request (https://github.com/Azure/azure-powershell/pull/3655) that will get rid of all of the XML help in this repository. This will move our partners and external contributors to update the markdown help from now on (and the XML help will be generated during the build).

As a result, I'm going to close this PR (since the XML file you are updating won't exist, and I regenerated all of the markdown files in the repository to make sure they were up to date with the latest changes in the code, so there may be merge conflicts), but please feel free to reopen this PR (or a new one) with the changes to the markdown when that PR is merged. I will also be including documentation that covers how to properly regenerated the markdown files using the `platyPS` module (which I will point you to).

Please let me know if you have any other questions!

**Pull request closed without merging**